### ISSUE TITLE
WQX

### ISSUE THREAD

**Issue by ksonda** [internetofwater/geoconnex.us#95]
Let's add them

 - which namespace? /wqx? /epa/wqx? 
 - do we want to pilot out some jsonld content for them and host them at info.geoconnex.us, or put that in a roadmap for WQP?

**Comment by jkreft-usgs** [comment]
Good Question!  The goal being to add all the monitoring locations that are part of the Exchange Network Water Quality Exchange? (e.g. the EPA part of the Water Quality Portal?)

**Comment by ksonda** [comment]
Ah yes. that raises another question. Should there be WQP? and do some `sameAs` -ing between the WQP and NWIS, WQP and Exchange Network, WQP and other ocntributory systems.

**Comment by dblodgett-usgs** [comment]
We certainly want PIDs for the water quality portal monitoring location pages and it would be good for those pages to link to reference features in info.geoconnex.us. We could create something akin to ref_gages that is more generally monitoring locations for that. The set of locations in the portal would be a good source to build that out.

**Comment by ksonda** [comment]
given how you've defined gages so far im not sure what would distinguish this reference set from ref_gages exactly.

**Comment by ksonda** [comment]
asnwering my own question, non-stream locations. got it.

**Comment by dblodgett-usgs** [comment]
Yeah -- "gage" is often through to measure flow but anything can be monitored at a gaging location. I could have called it reference_stream_monitoring_locations but that seemed a little overdone ;)

**Comment by ksonda** [comment]
mkay, so what im hearing, for now, is we're going to make two sets of PIDS, 

- /ref/monitoring-sites?/
- /wqp/sites?  /nwqmc/wqp/sites?

where the monitoring-sites are in info.geoconnex.us and subjectOf the WQP PIDs, and for now they will be the same thing. Maybe the monitoring-sites that are also in NWIS are also subjectOf the /usgs/monitoring-location PIDs.

**Comment by dblodgett-usgs** [comment]
Or we do:
- /ref/wells/ in addition to /ref/gages/ 
- /wqp/sites/

Where wqp sites are sameAs USGS and/or EPA and/or state data sources and are about / subjectOf with the appropriate reference feature.

**Comment by ksonda** [comment]
WQP has all kinds of weird other site types like distribution system samples and internal facility samples and such though, which eventually should go in probably. 

I see why your way is conceptually cleaner, its just more long term maintenance work.
we can bite off ref/wells as a start though and use WQP wells as a base, we've been meaning to do that anyway.

**Comment by dblodgett-usgs** [comment]
I like the idea of having a super set of reference locations that are either special or sameAs to the gage or well set.

**Comment by ksonda** [comment]
mkay so we're saying 3 levels of stuff now?

- /ref/monitoring-sites
-^sameAs
- /ref/gages, /ref/wells, and later on /ref/facilities etc.
- ^subjectOf
- /wqp/sites, usgs/monitoring-locations/, state-agency/gages, etc.

**Comment by dblodgett-usgs** [comment]
The sameAs isn't really necessary as a level if we want to have a ref/foobarbaz/ for things that don't fall in a major category.

**Comment by ksonda** [comment]
Im thinking if we shunt everything "not cateogrized" into /ref/foobar,  then we have to mint new PIDs that are sameAs the PIDs in /ref/foobar if a new category justifies itself down the line. 

Vs. 3 levels, where the top level can be part of a workflow that is auto-maintained from the set of location type namespaces as they emerge.

**Comment by ksonda** [comment]
we can just make /wqp/ for now just so it's there. regex?

**Comment by dblodgett-usgs** [comment]
What do you think @jkreft-usgs ? Site pages are like : https://www.waterqualitydata.us/provider/NWIS/USGS-AL/USGS-02339215/

**Comment by jkreft-usgs** [comment]
I think that's fine for now, and then we can figure out more later.

**Comment by ksonda** [comment]
For now, the YOURLs-based PID server can really only handle regex for the "basename" of a URL. E.g. only USGS-02339215 would get passed through from something like https://geoconnex.us/wqp/sites/NWIS/USGS-AL/USGS-02339215

Having @webb-ben look at this. 

Short term workaround could be a series of regex PIDs that are based on data source and provider

^wqp/sites/NWIS/USGS-AL/{id regex} --> https://www.waterqualitydata.us/provider/NWIS/USGS-AL/{id regex match}

^wqp/sites/STORET/NC-WQX/{id regex} --> https://www.waterqualitydata.us/provider/STORET/NC-WQX/{id regex match}

and so on. And maybe in WQP case this is an appropriate way to do things anyway, since PIDs are supposed to be organized around "organization" in some sense.

**Comment by ksonda** [comment]
lol nvm. https://github.com/USGS-R/dataRetrieval/issues/434#issuecomment-522230798

Ill prioritize fixing pid server

**Comment by ksonda** [comment]
apologies for earlier today when the hu10 was down @dblodgett-usgs , I hope that wasn't a bad hiccup for something you were trying to show off or test.

However, fixing that error resulted in us figuring out some other stuff, resulting in this pattern functioning for WQP:

http://geoconnex.us/wqp/sites/([a-zA-Z0-9/-]+)$

**Comment by ksonda** [comment]
This cool with you @jkreft-usgs ?

**Comment by ksonda** [comment]
@jkreft-usgs I've noticed in the NLDI now that WQP uri links are broken. URLs are like https://www.waterqualitydata.us/data/provider/STORET/WIDNR_WQX/WIDNR_WQX-10017576/ when they need to be like

https://www.waterqualitydata.us/provider/STORET/WIDNR_WQX/WIDNR_WQX-10017576/

Alternatively could replace with https://geoconnex.us/wqp/STORET/WIDNR_WQX/WIDNR_WQX-10017576/ or whatever :)

**Comment by jkreft-usgs** [comment]
blerg, that's a bug on the WQP side!  I'll get a ticket in to fix it.

**Comment by C-Loftus** [comment]
I believe we have the latest water quality portal in the geoconnex graph. I think this issue is a bit old now and is likely stale so I will close to keep things up to date. If anyone wants any updates on this integration, please feel free to create a new issue with latest info and I'm happy to look at anything.
